### ISSUE TITLE
Filter Predicate executed twice when using Eval.now

### ISSUE THREAD

**Issue by jlorenzen** [aol/cyclops-react#730]
When using filter with `Eval.now` and a `Maybe.orElseThrow`, the filter predicate is executed twice, which I think is unexpected (or at least it was for me).

```java
Eval.now("50")
    .filter(it -> {
        System.out.println("filter");
        return true;
    })
    .orElseThrow(() -> new RuntimeException("bad"));
```

This will print
```
filter
filter
```

* When I use `get` instead of `orElseThrow` it only executes the filter predicate once.
* When I use `Eval.later` instead of `Eval.now` it only executes the filter predicate once.

**Comment by johnmcclean** [comment]
Thanks @jlorenzen , it looks like the conversion to a Maybe is also converting the Eval type to a memoized Later and that is somehow involved in causing this bug. There is no need for this,  simplifying either the toMaybe and / or filter methods to something specific to the Eval type will fix this issue (and make the stacktraces much smaller / cleaner also).

**Comment by johnmcclean** [comment]
Released in 2.1.1
